File: solutions/julia/difference-of-squares/1/difference-of-squares.jl
"Square the sum of the first `n` positive integers"
function square_of_sum(n)
    sum = 0
    for x in 1:n
        sum = sum + x
    end
    return sum * sum 
end

"Sum the squares of the first `n` positive integers"
function sum_of_squares(n)
    sum = 0
    for x in 1:n
       sum = sum + (x * x)
    end
    return sum
end

"Subtract the sum of squares from square of the sum of the first `n` positive ints"
function difference(n)
    diff = square_of_sum(n) - sum_of_squares(n)
    return diff
end

File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number):
    sum = 0
    for x in range(1, number + 1):
        sum += x
    return sum ** 2


def sum_of_squares(number):
    sum = 0
    for x in range(1, number + 1):
        sum += x ** 2
    return sum


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)
